server/utils/user/UserFactory.ts

```
@@ -0,0 +1,207 @@
+                    };
+                }
+            }
+
+            // If the user ever finds himself in a weird state where the profile doesn't exist but a dbUserID does
+            // we recreate a userProfile with a new id
+            const newUser = {
+                id: uuid(),
+                username: username,
+                lastLogin: new Date().toISOString(),
+                createdAt: new Date().toISOString(),
+                usernameSetAt: new Date().toISOString(),
+                preferences: { cardback: null },
+            };
+
+            // Create OAuth link
+            await this.dynamoDbService.saveOAuthLink(provider, providerId, newUser.id);
+            // Save the user profile
+            await this.dynamoDbService.saveUserProfile(newUser);
+            // Create email link if email is available
+            if (!email) {
+                throw new Error(`Email not found for user ${newUser.id}`);
+            }
+            await this.dynamoDbService.saveEmailLink(email, newUser.id);
+            logger.info(`Created new user: ${newUser.id} (${username}) with ${provider} authentication`);
+            return {
+                id: newUser.id,
+                username: newUser.username,
+            };
+        } catch (error) {
+            // This should never happen but if it does its so we don't create two accounts
+            if (error.name === 'ConditionalCheckFailedException') {
+                // This could happen if there's a race condition while creating user records
+                // or if the unique constraints we're enforcing are violated for other reasons
+                logger.error(`DynamoDB conditional check failed during user authentication: ${error}`);
+                return null;
+            }
+            // This catches both JWT verification errors and database errors
+            logger.error('Authentication error:', error);
+            return null;
```

I could go either way on this one but still feels like it should throw to indicate that an unexpected failure happened, and then let the caller deal with it by catching if they want to